Fix another inlet parameter check for scalar equations

GiudGiud opened the pull request:
refs #22774

what we have checks
N_inlet_types = N_boundaries

but if we have 2 scalars we ll have
2 * N_inlet_types because the inlet types might be different for each scalar

GiudGiud committed:
Fix inlet parameter checks for scalar equations:
number of inlet types should be N_scalar * N_inlet to allow for multiple scalars for each inlet
refs #22774

grmnptr commented:
You might need to update this guy too `navier_stokes/test:finite_volume/ins/action/errors.passive-scalar-inlet-error`.

grmnptr commented on modules/navier_stokes/src/actions/NSFVAction.C:
```diff
@@ -2596,11 +2596,11 @@ NSFVAction::checkBoundaryParameterErrors()
     {
-      checkSizeFriendParams(_inlet_boundaries.size(),
+      checkSizeFriendParams(_inlet_boundaries.size() * _passive_scalar_names.size(),
                             _passive_scalar_inlet_types.size(),
```
Shouldn't this be a `vector<vector<string>` param too to make sure there is a one-to-one correspondence between this guy and the `_passive_scalar_inlet_function`?

GiudGiud replied:
we have it as a MultiMooseEnum which has a nicer param list
there's a trade off until we support std::vector<MultiMooseEnum> which would offer both

grmnptr replied:
We can still use the MultiMooseEnum to check the input, but in the constructor. However at the moment we have something like this for the inlet functions:

```
passive_scalar_inlet_function = `inlet_1_var_1 inlet_2_var_1; inlet_1_var2 inlet_2_var2`
```

and I think it would look bad with the multimoose enum:

```
passive_scalar_inlet_types = `inlet_1_var_1 inlet_2_var_1 inlet_1_var2 inlet_2_var2`
```
 I think we need to make that a vector of vectors too and loop through all of them.

Also this will not solve the problem which is this:

```
or (unsigned int name_i = 0; name_i < _passive_scalar_names.size(); ++name_i)
  {
    unsigned int flux_bc_counter = 0;
    for (unsigned int bc_ind = 0; bc_ind < _inlet_boundaries.size(); ++bc_ind)
    {
      if (_passive_scalar_inlet_types[bc_ind] == "fixed-value")
      {
```

So we will need:
```
if (_passive_scalar_inlet_types[name_i][bc_ind] == "fixed-value")
      {
``` 

I can push this later today if you are short on time.

GiudGiud replied:
in the documentation page for the action, if we do not use a multiMooseEnum for the parameter, we will not be telling the user which options are accepted

grmnptr replied:
Well, we have been doing this at other instances too:

```
params.addParam<std::vector<std::vector<std::string>>>(
      "friction_types", "The types of friction forces for every block in 'friction_blocks'.");
```

We can still add the options for the parameters manually. Or 
you can do one index [name_i*num_bcs + bc_ind] (or the other way around) but the code I linked needs to be changed either way.

GiudGiud replied:
well this isnt the best. There s only two options here there though

GiudGiud replied:
so it makes it easier to specify the types correctly

GiudGiud committed:
Address review:
- correctly index into scalar inlet types when selecting BCs
- refer to scalar BC type instead of energy BC type
Refs #22774
